packages/@react-aria/slider/src/useSliderThumb.ts

```diff
@@ -77,44 +77,77 @@ export function useSliderThumb(
   stateRef.current = state;
   let reverseX = direction === 'rtl';
   let currentPosition = useRef<number>(null);
+
+  let {keyboardProps} = useKeyboard({
+    onKeyDown(e) {
+      let {
+        getThumbMaxValue,
+        getThumbMinValue,
+        pageSize
+      } = stateRef.current;
+      // these are the cases that useMove or useSlider don't handle
+      if (!/^(PageUp|PageDown|Home|End)$/.test(e.key)) {
+        e.continuePropagation();
+        return;
+      }
+      // same handling as useMove, stopPropagation to prevent useSlider from handling the event as well.
+      e.preventDefault();
+      // remember to set this so that onChangeEnd is fired
+      state.setThumbDragging(index, true);
+      switch (e.key) {
+        case 'PageUp':
+          stateRef.current.incrementThumb(index, pageSize);
+          break;
+        case 'PageDown':
+          stateRef.current.decrementThumb(index, pageSize);
+          break;
+        case 'Home':
+          state.setThumbValue(index, getThumbMinValue(index));
+          break;
+        case 'End':
+          state.setThumbValue(index, getThumbMaxValue(index));
+          break;
+      }
+      state.setThumbDragging(index, false);
+    }
+  });
+
   let {moveProps} = useMove({
-    onMoveStart({pointerType}) {
+    onMoveStart() {
       currentPosition.current = null;
       // Don't start dragging for keyboard events unless the value has changed.
```

Comment:
This comment is only with the `pointerTypo === 'keyboard'` conditional added below.